pkg/cli/config/config.go

```diff
@@ -96,6 +98,52 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
 	return bpConfig, nil
 }
 
+func getProxyURL(bpConfig BackplaneConfiguration, proxyURLConfig any) string {
+	// as part of OSD-19785, proxy-url in config file is converted to []string from string to allow multiple proxies.
+	// For backward compatibility, get proxy-url as 'any' and check if it is still string (set in past) or array
+	// if string, return it else if array, return the first working proxy
+	switch t := proxyURLConfig.(type) {
+	case []interface{}:
+		logger.Infof("proxy-url array found in config file: %s ", t)
+		return getFirstWorkingProxyURL(bpConfig, proxyURLConfig.([]interface{}))
+	case interface{}:
+		logger.Warn("Change proxy-url in config file to an array. Multiple proxy-urls are now supported. First working proxy-url from the array will be used")
+		return proxyURLConfig.(string)
+	default:
+		logger.Warnf("Unrecognizable value type for proxy-url in config file: %s", t)
+		return ""
+	}
+}
+
+func getFirstWorkingProxyURL(bpConfig BackplaneConfiguration, s []interface{}) string {
+	bpURL := bpConfig.URL + "/healthz"
+	logger.Infof("bpUrl: %s", bpURL)
+	for _, p := range s {
+		logger.Infof("proxy to check: %s", p)
+		if isProxyWorking(bpURL, p.(string)) {
+			return p.(string)
+		}
+	}
+	return ""
```

Comment:
I guess you may need to return an error if any of the configured proxy is not functioning

Reply:
Done